tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -2135,8 +2135,8 @@ public function test_is_woopay_enabled_returns_true() {
 		$this->wcpay_gateway->update_option( 'platform_checkout', 'yes' );
 		$this->assertTrue( $this->woopay_utilities->should_enable_woopay( $this->wcpay_gateway ) );
 
-		// This will return TRUE because woopay_utilities->should_enable_woopay() will return true.
-		$this->assertTrue( $this->payments_checkout->get_payment_fields_js_config()['isWooPayEnabled'] );
+		// This will return false because woopay_utilities->should_enable_woopay_on_cart_or_checkout() will return false.
+		$this->assertFalse( $this->payments_checkout->get_payment_fields_js_config()['isWooPayEnabled'] );
```

Comment:
This is definitely an odd test case now with those conflicts on true / false. I was thinking one way to handle it is if we set a current user (`wp_set_current_user( 1 )`) and can set the `is_checkout` function to be true (I couldn't figure this one out quickly), then I think we would get our desired result. I wouldn't be a fan of it, but I suppose we could also mock `should_enable_woopay_on_cart_or_checkout` as we're primarily checking the JS config result?

Reply:
Tweaked the test to make sure we are checking the expected value here https://github.com/Automattic/woocommerce-payments/pull/6509/commits/0c94bce43884a73620d334af986b4ff84b0d3008.

Reply:
Thanks so much for taking a look at that @asumaran I was still a bit hesitant to use a mock there, especially since that utility function is used in other tests. I came up with the following diff that seems to get the test passing without a mock. Basically I just set some conditions that would allow the result to be true (on the checkout page, logged in user). Let me know what you think!

```diff
diff --git a/tests/unit/test-class-wc-payment-gateway-wcpay.php b/tests/unit/test-class-wc-payment-gateway-wcpay.php
index 59549b79..aa7ece07 100644
--- a/tests/unit/test-class-wc-payment-gateway-wcpay.php
+++ b/tests/unit/test-class-wc-payment-gateway-wcpay.php
@@ -82,10 +82,9 @@ class WC_Payment_Gateway_WCPay_Test extends WCPAY_UnitTestCase {
 	/**
 	 * WooPay_Utilities instance
 	 *
-	 * @var WooPay_Utilities|PHPUnit_Framework_MockObject_MockObject
-	 *
+	 * @var WooPay_Utilities
 	 */
-	private $mock_woopay_utilities;
+	private $woopay_utilities;
 
 	/**
 	 * Session_Rate_Limiter instance.
@@ -179,11 +178,11 @@ class WC_Payment_Gateway_WCPay_Test extends WCPAY_UnitTestCase {
 			$this->mock_dpps
 		);
 
-		$this->mock_woopay_utilities = $this->createMock( WooPay_Utilities::class );
+		$this->woopay_utilities = new WooPay_Utilities();
 
 		$this->payments_checkout = new WC_Payments_Checkout(
 			$this->wcpay_gateway,
-			$this->mock_woopay_utilities,
+			$this->woopay_utilities,
 			$this->mock_wcpay_account,
 			$this->mock_customer_service
 		);
@@ -2142,18 +2141,15 @@ class WC_Payment_Gateway_WCPay_Test extends WCPAY_UnitTestCase {
 	}
 
 	public function test_is_woopay_enabled_returns_true() {
-		$this->mock_woopay_utilities
-			->method( 'should_enable_woopay_on_cart_or_checkout' )
-			->willReturn( true );
-		$this->mock_woopay_utilities
-			->method( 'should_enable_woopay' )
-			->willReturn( true );
-
 		$this->mock_cache->method( 'get' )->willReturn( [ 'platform_checkout_eligible' => true ] );
 		$this->wcpay_gateway->update_option( 'platform_checkout', 'yes' );
+		wp_set_current_user( 1 );
+		add_filter( 'woocommerce_is_checkout', '__return_true' );
 
-		$this->assertTrue( $this->mock_woopay_utilities->should_enable_woopay( $this->wcpay_gateway ) );
+		$this->assertTrue( $this->woopay_utilities->should_enable_woopay( $this->wcpay_gateway ) );
 		$this->assertTrue( $this->payments_checkout->get_payment_fields_js_config()['isWooPayEnabled'] );
+
+		remove_filter( 'woocommerce_is_checkout', '__return_true' );
 	}
 
 	public function test_should_use_stripe_platform_on_checkout_page_not_woopay_eligible() {
@@ -2178,7 +2174,7 @@ class WC_Payment_Gateway_WCPay_Test extends WCPAY_UnitTestCase {
 
 		$payments_checkout = new WC_Payments_Checkout(
 			$mock_wcpay_gateway,
-			$this->mock_woopay_utilities,
+			$this->woopay_utilities,
 			$this->mock_wcpay_account,
 			$this->mock_customer_service
 		);
@@ -2323,7 +2319,7 @@ class WC_Payment_Gateway_WCPay_Test extends WCPAY_UnitTestCase {
 
 		$this->payments_checkout = new WC_Payments_Checkout(
 			$this->wcpay_gateway,
-			$this->mock_woopay_utilities,
+			$this->woopay_utilities,
 			$this->mock_wcpay_account,
 			$this->mock_customer_service
 		);
```

Reply:
Cool! That's a much better approach I think. Can you commit the changes or can I do it?

Reply:
Sure I can commit that now!